### PR TITLE
docs: fix version change

### PR DESCRIPTION
**Pull request by ybzjsxh**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [x] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Perfermance optimization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |         |
| 🇨🇳 Chinese |   文档版本跳转问题   |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: site/theme/template/Layout/Header/index.tsx [hunk 1]
```diff
@@ -146,6 +146,12 @@ class Header extends React.Component<HeaderProps, HeaderState> {
   handleVersionChange = (url: string) => {
     const currentUrl = window.location.href;
     const currentPathname = window.location.pathname;
+    if (/overview/.test(currentPathname)) {
+      window.location.href = currentUrl
+        .replace(window.location.origin, url)
+        .replace(/\/components\/overview/, '/docs/react/introduce');
+      return;
+    }
```
Comment (afc163) [inline site/theme/template/Layout/Header/index.tsx:154]: 判断一下版本。

Comment (ybzjsxh) [reply site/theme/template/Layout/Header/index.tsx:155]: 好的大佬，加了09x，010x版本的判断，您看看

Comment (afc163) [reply site/theme/template/Layout/Header/index.tsx:155]: 除了 1.x 2.x 3.x 都可以加上，只有 4.0 有这个页面。

Comment (ybzjsxh) [reply site/theme/template/Layout/Header/index.tsx:155]: 嗯嗯，我这里的改动是0.11.x, 0.12.x,1.x, 2.x, 3.x的跳/docs/react/introduce, 0.9x, 0.10.x跳/docs/introduce这样，下拉框里的都试了下能正常跳转
image

Comment (afc163) [reply site/theme/template/Layout/Header/index.tsx:155]: 多加个判断，未来 5.x 跳 4.x 就不用修改 url 了。

Comment (ybzjsxh) [reply site/theme/template/Layout/Header/index.tsx:155]: 好的，现在3x及以下才会修改url了

**Pull request merged**